internal/controller/apiextensions/claim/configurator.go

```diff
@@ -183,82 +179,94 @@ func filter(in map[string]any, keys ...string) map[string]any {
+
+	// We refuse to 're-bind' a claim that is already bound to a different
+	// composite resource.
+	if existing != nil && !equal {
+		return errors.New(errBindClaimConflict)
 	}
-	ucp, ok := cp.(*composite.Unstructured)
-	if !ok {
-		return nil
+
+	cmPatch.SetResourceReference(proposed)
+
+	// If existing claim has the status set,
+	// copy the conditions to the patch
+	if s, ok := cm.Object["status"]; ok {
+		fs, ok := s.(map[string]any)
+		if !ok {
+			return errors.Wrap(errors.New(errUnsupportedSrcObject), errMergeClaimStatus)
+		}
+		cmPatch.Object["status"] = keep(fs, "conditions")
+	} else {
+		cmPatch.Object["status"] = map[string]any{}
 	}
 
-	if err := merge(ucm.Object["status"], ucp.Object["status"],
+	// merge from the composite status everything
+	// except conditions and connectionDetails
+	if err := merge(cmPatch.Object["status"], cp.Object["status"],
 		// Status fields from composite overwrite non-empty fields in claim
 		withMergeOptions(mergo.WithOverride),
 		withSrcFilter(xcrd.GetPropFields(xcrd.CompositeResourceStatusProps())...)); err != nil {
 		return errors.Wrap(err, errMergeClaimStatus)
 	}
 
-	if err := c.client.Status().Update(ctx, cm); err != nil {
-		return errors.Wrap(err, errUpdateClaimStatus)
-	}
-
 	// Propagate the actual external name back from the composite to the
 	// claim if it's set. The name we're propagating here will may be a name
 	// the XR must enforce (i.e. overriding any requested by the claim) but
 	// will often actually just be propagating back a name that was already
 	// propagated forward from the claim to the XR during the
 	// preceding configure phase.
 	if en := meta.GetExternalName(cp); en != "" {
-		meta.SetExternalName(cm, en)
-	}
-
-	// We want to propagate the composite's spec to the claim's spec, but
-	// first we must filter out any well-known fields that are unique to
-	// composites. We do this by:
-	// 1. Grabbing a map whose keys represent all well-known composite fields.
-	// 2. Deleting any well-known fields that we want to propagate.
-	// 3. Filtering OUT the remaining map keys from the composite's spec so
-	// that we end up adding only the well-known fields to the claim's spec.
-	wellKnownCompositeFields := xcrd.CompositeResourceSpecProps()
-	for _, field := range xcrd.PropagateSpecProps {
-		delete(wellKnownCompositeFields, field)
+		meta.SetExternalName(cmPatch, en)
 	}
 
 	// CompositionRevision is a special field which needs to be propagated
 	// based on the Update policy. If the policy is `Automatic`, we need to
 	// overwrite the claim's value with the composite's which should be the
 	// `currentRevision`
-	if cp.GetCompositionUpdatePolicy() != nil && *cp.GetCompositionUpdatePolicy() == xpv1.UpdateAutomatic {
-		cm.SetCompositionRevisionReference(cp.GetCompositionRevisionReference())
+	if cp.GetCompositionUpdatePolicy() != nil &&
+		*cp.GetCompositionUpdatePolicy() == xpv1.UpdateAutomatic && cp.GetCompositionRevisionReference() != nil {
```

Comment:
With this added condition `cp.GetCompositionRevisionReference() != nil` here, we will only set `compositionRevisionReference` once in the beginning instead of keeping it always up to date when the policy Automatic.

Reply:
The semantic is not changed - it is going to keep mirroring the revision reference. 

`cp.GetCompositionRevisionReference() != nil` was needed to introduce here, in order to prevent creation of invalid claim patches. Namely, 2nd claim reconciliation can happen before the composite got reconciled and the revision reference can be still `nil`. Without the new condition, claim patch would be invalid. If a field is nil, then it should not be a part of patch at all.

Reply:
ah ok, it seems like I read it as `if cp.GetCompositionRevisionReference() == nil` for some reason 🤦